nutils/function.py

```diff
@@ -192,6 +192,16 @@ def lower(self, points_shape: _PointsShape, transform_chains: _TransformChainsMa
   def as_evaluable_array(self) -> evaluable.Array:
     return self.lower((), {}, {})
 
+  def __index__(self):
+    if self.arguments:
```

Comment:
I think this should be `if self.arguments or self.spaces:`.

Reply:
fixed